reference/preconditioner/batch_scalar_jacobi.hpp

```diff
@@ -0,0 +1,138 @@
+// SPDX-FileCopyrightText: 2017 - 2024 The Ginkgo authors
+//
+// SPDX-License-Identifier: BSD-3-Clause
+
+#ifndef GKO_REFERENCE_PRECONDITIONER_BATCH_SCALAR_JACOBI_HPP_
+#define GKO_REFERENCE_PRECONDITIONER_BATCH_SCALAR_JACOBI_HPP_
+
+
+#include "core/base/batch_struct.hpp"
+#include "core/matrix/batch_struct.hpp"
+
+
+namespace gko {
+namespace kernels {
+namespace host {
+namespace batch_preconditioner {
+
+
+/**
+ * (Scalar) Jacobi preconditioner for batch solvers.
+ */
+template <typename ValueType, typename IndexType = int>
+class ScalarJacobi final {
+public:
+    /**
+     * The size of the work vector required in case of dynamic allocation.
+     */
+    static int dynamic_work_size(const int nrows, int) { return nrows; }
+
+    /**
+     * Sets the input and generates the preconditioner by storing the inverse
+     * diagonal entries in the work vector.
+     *
+     * @param mat  Matrix for which to build a Jacobi preconditioner.
+     * @param work  A 'work-vector', used here to store the inverse diagonal
+     *              entries. It must be allocated with at least the amount
+     *              of memory given by work_size or dynamic_work_size.
+     */
+    void generate(size_type,
+                  const gko::batch::matrix::ell::batch_item<
+                      const ValueType, const IndexType>& mat,
+                  ValueType* const work)
+    {
+        work_ = work;
+        for (int i = 0; i < mat.num_rows; i++) {
```

Comment:
maybe initialize `work[i] = 1`, to handle non-existing diag gracefully (same for csr)

Reply:
now the check for the diagonal isn't necessary anymore.